share/rocm/cmake/ROCMWrapper.cmake

```diff
@@ -0,0 +1,99 @@
+# ######################################################################################################################
+# Copyright (C) 2022 Advanced Micro Devices, Inc.
+# ######################################################################################################################
+
+set(ROCM_SOURCE_DIR "${CMAKE_CURRENT_LIST_DIR}" CACHE INTERNAL "Path to rocm-cmake files.")
```

Comment:
What we would actually want to use is `CMAKE_CURRENT_FUNCTION_LIST_DIR`, so that the function can refer to the directory its file is located in, but that wasn't introduced until CMake 3.17. This is the structure that variable was intended to replace. Another option would be to use `ROCM_DIR`, which should be set when the `ROCM` package is found via `find_package`, but that would probably be messier.

I do like the suggestion of having the variable for the template header, it's more flexible and clearer in purpose.